subsys/bluetooth/host/l2cap.c

```diff
@@ -1652,7 +1656,7 @@ static void le_ecred_conn_rsp(struct bt_l2cap *l2cap, uint8_t ident,
 	}
 
 	if (ecred_cb && ecred_cb->ecred_conn_rsp) {
-		ecred_cb->ecred_conn_rsp(conn, result, attempted, succeeded);
+		ecred_cb->ecred_conn_rsp(conn, result, attempted, succeeded, psm);
```

Comment:
Since the `ecred_conn_rsp` is an internal callback we haven't really documented it which is fine I suppose, but the it may not be clear that the `psm` is the `psm` of the last channel handled. 

Since we are working on multiple channels, perhaps the callback should contain an array of { psm, success}, e.g.

```c
struct ecred_conn_rsp_res {
    bool success;
    uint16_t psm;
}

void (*ecred_conn_rsp)(struct bt_conn *conn, uint16_t result, struct ecred_conn_rsp_res rsp_res[], size_t rsp_cnt);
```

What do you think? Or is the PSM of the last channel all we need?

Reply:
All the channels have the same `ident`, so they come from the same request and thus have the same `psm`.

Though I don't really like these callbacks so I am open to changing them.

Reply:
Ah, in that case the `psm` assignment should probably be moved outside of the `while` loops (no reason to assign it multiple times if it will be the same). Perhaps we should add an `__ASSERT` to verify that the channel do indeed have the same PSM then though?

Reply:
Maybe better to add the assert in `l2cap_ecred_conn_req()` instead/as well? There the PSM of the first channel is used for the request.